### PR TITLE
Finished_footprints_for_precharge

### PR DESCRIPTION
**Pull request by Derek709**
New layout footprints for Precharge. Designed form specifications made on old Diptrace file.

### CODE REVIEW

**Comment by Derek709** [comment]
> It looks like you put things into the KiCad default library files. Did you take these from a more recent version of the official KiCad libraries? If not they will be overwritten the next time those are updated and will be lost. In that case move them to the ISC libraries.

I did not. How can I put them into the non-default libraries folder, so I can push it?

**Comment by kamatar2** [comment]
You can copy the files you meant to put in our libraries and paste them in schematic and layout.pretty respectively, then delete the symbols and footprints from the kicad defaults

**Comment by Derek709** [comment]
> Now it looks like you've moved a couple of files out of the kicad libraries that are a part of the kicad libraries.
> 
> Specifically:
> 
> `footprint_kicad/Resistor_THT.pretty/R_Axial_Power_L38.0mm_W9.0mm_P45.72mm.kicad_mod → layout.pretty/R_Axial_Power_L38.0mm_W9.0mm_P45.72mm.kicad_mod`
> 
> `footprint_kicad/Converter_DCDC.pretty/Converter_DCDC_RECOM_R-78S-0.1_THT.kicad_mod → layout.pretty/Converter_DCDC_RECOM_R-78S-0.1_THT.kicad_mod`
> 
> `footprint_kicad/Package_TO_SOT_SMD.pretty/SOT-23.kicad_mod → layout.pretty/SOT-23.kicad_mod`

Gotcha. I can get that fixed.

File: schematic_kicad/Isolator.lib [hunk 1]
```diff
@@ -1315,6 +1315,26 @@ X VCC 8 300 300 100 L 50 50 1 1 P
 ENDDRAW
 ENDDEF
 #
+# I2C_Isolator
```
Comment (jtmullen) [inline schematic_kicad/Isolator.lib:1318]: This should be in the schematic folder, not schematic_kicad

**Comment by Derek709** [comment]
I made the changes, but I was confused on the Relay comments you left. What do you mean by document link and keywords? I also don't understand the comment about the 2nd ref des.

**Comment by jtmullen** [comment]
> I also don't understand the comment about the 2nd ref des.

You should read the KiCad library conventions on Footprint Layer requirements (section F5) and specifically [F5.2 on fabrication layers](https://kicad-pcb.org/libraries/klc/F5.2/)



> What do you mean by document link and keywords?

In the footprint properties menu you need to update those two fields. The menu is in the top bar and File->Properties
image

**Comment by kamatar2** [comment]
### Relay_SPST_Omron_G6DN
- It looks like you still need the ref des on f.fab layer. and Your pin 1 is not centered at (0,0)
- Your ref des is hardcoded rather than being ref**

**Comment by kamatar2** [comment]
### I2C Isolator Symbol
- Still in a Kicad default symbol library
- Add description, keywords and datasheet in the symbol properties

**Comment by kamatar2** [comment]
You also still need to remove all the unnecessary footprints you created from the pull request and change the isolators lib back to schematic_kicad. See the files changed tab for an easy list of what I'm talking about

**Comment by Derek709** [comment]
what is the non kicad symbol library we use? I couldn't find it.

**Comment by Derek709** [comment]
@kamatar2

**Comment by kamatar2** [comment]
Our ICs library is probably best for the isolator symbol

**Comment by Derek709** [comment]
I think I might have accidentally removed the relay schematic library. How can I backdate to a previous version or download that from a previous version to get it back?

**Comment by kamatar2** [comment]
A couple things you can do. One of the easiest is just to checkout an old commit on the same branch and copy the footprint from there into a non git folder. Then you can jump back to the head of the branch and just paste it. I've done this before because it involves the least interaction with git commands

**Comment by Derek709** [comment]
@jtmullen So I don't get the same footprint properties menu that you showed earlier in this pull request. It's similar, but lists Footprint name, Description, and Keywords rather than Document Link. I've made the change in these fields, so could that not show up in whatever menu you are seeing? I can't find a footprint properties under file, only in edit in the top bar. Also, the pad numbers should be correct as should the Refdes on F.Fab.

**Comment by jtmullen** [comment]
>  I don't get the same footprint properties menu that you showed earlier 

I've learned that it has been removed in newer versions of KiCad, so you can ignore the need for that.

**Comment by Derek709** [comment]
Noticed the changes to Relay footprint didn't save. Hopefully that's fine now.

**Comment by Derek709** [comment]
I believe you mentioned ignoring the need for a document link, because that was only on older versions of KiCad.

File: layout.pretty/SOIC-8_5.85x7.4mm_P1.27mm.kicad_mod [hunk 2]
```diff
@@ -1,8 +1,8 @@
 (module SOIC-8_5.85x7.4mm_P1.27mm (layer F.Cu) (tedit 5CB406ED)
-  (fp_text reference REF** (at -4.475 -3.875) (layer F.SilkS)
+  (fp_text reference U1 (at -2.159 4.318 90) (layer F.SilkS)
```
Comment (jtmullen) [inline layout.pretty/SOIC-8_5.85x7.4mm_P1.27mm.kicad_mod:2]: This ref des is wrong now

File: layout.pretty/SOIC-8_5.85x7.4mm_P1.27mm.kicad_mod [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
     (effects (font (size 1 1) (thickness 0.15)))
   )
-  (fp_text value SOIC-8_5.85x7.4mm_P1.27mm (at -0.075 4.3) (layer F.Fab)
+  (fp_text value Si8602AC-B-IS (at -0.075 4.3) (layer F.Fab)
```
Comment (jtmullen) [inline layout.pretty/SOIC-8_5.85x7.4mm_P1.27mm.kicad_mod:5]: Why did you change this?

File: schematic/ICs.dcm [hunk 4]
```diff
@@ -180,6 +180,12 @@ K DCDC DC-DC converter
 F https://recom-power.com/pdf/Econoline/RSO.pdf
 $ENDCMP
 #
+$CMP Si8602-Isolator
```
Comment (jtmullen) [inline schematic/ICs.dcm:183]: Change the name of this to just Si8602

File: layout.pretty/Relay_SPST_Omron_G6DN.kicad_mod [hunk 5]
```diff
@@ -0,0 +1,30 @@
+(module Relay_SPST_Omron_G6DN (layer F.Cu) (tedit 5F2890A6)
+  (descr "SPST Slim Power Relay for 5 A Switching")
+  (tags "Relay SPST Omron 4Pin")
+  (fp_text reference K1 (at 20.4397 1.16824 -90) (layer F.SilkS)
```
Comment (jtmullen) [inline layout.pretty/Relay_SPST_Omron_G6DN.kicad_mod:4]: The Refdes here should not be K1 or anything specific. It should be `**REF` (change this in footprint properties)

**Review by jtmullen — COMMENTED** [review]
Oops, wrong PR.
